diff --git a/index.d.ts b/index.d.ts
index b59810ce0fcaaf647830359010987348e11b0479..0f9d24ab390ac12010b671bc8aab02909e93fa89 100644
--- a/index.d.ts
+++ b/index.d.ts
@@ -39,7 +39,7 @@ interface USBConnectionEventInit extends EventInit {
 
 declare class USBConfiguration {
     readonly configurationValue: number;
-    readonly configurationName?: string | undefined;
+    readonly configurationName: string | null;
     readonly interfaces: USBInterface[];
 }
 
@@ -57,14 +57,14 @@ declare class USBAlternateInterface {
     readonly interfaceClass: number;
     readonly interfaceSubclass: number;
     readonly interfaceProtocol: number;
-    readonly interfaceName?: string | undefined;
+    readonly interfaceName: string | null;
     readonly endpoints: USBEndpoint[];
 }
 
 declare class USBInTransferResult {
     constructor(status: USBTransferStatus, data?: DataView);
     readonly data?: DataView | undefined;
-    readonly status?: USBTransferStatus | undefined;
+    readonly status: USBTransferStatus;
 }
 
 declare class USBOutTransferResult {
@@ -76,7 +76,7 @@ declare class USBOutTransferResult {
 declare class USBIsochronousInTransferPacket {
     constructor(status: USBTransferStatus, data?: DataView);
     readonly data?: DataView | undefined;
-    readonly status?: USBTransferStatus | undefined;
+    readonly status: USBTransferStatus;
 }
 
 declare class USBIsochronousInTransferResult {
@@ -140,10 +140,10 @@ declare class USBDevice {
     readonly deviceVersionMajor: number;
     readonly deviceVersionMinor: number;
     readonly deviceVersionSubminor: number;
-    readonly manufacturerName?: string | undefined;
-    readonly productName?: string | undefined;
-    readonly serialNumber?: string | undefined;
-    readonly configuration?: USBConfiguration | undefined;
+    readonly manufacturerName: string | null;
+    readonly productName: string | null;
+    readonly serialNumber: string | null;
+    readonly configuration: USBConfiguration | null;
     readonly configurations: USBConfiguration[];
     readonly opened: boolean;
     open(): Promise<void>;
